Uppy 2.0.0 release post (#3100)
transloadit/uppy: website/src/_posts/2021-08-30.31.md

- [That's it](#That’s-it)

## Highlights since 1.0

A lot of things have happened since we released Uppy 1.0 in April of 2019. In case you have missed some of our 1.x releases, here is a small overview. We have added:

- Official integrations for [Vue](/blog/2020/11/1.23/) ([3](/blog/2021/04/1.27/)), [Angular](/blog/2021/07/1.30/), and [Svelte](/blog/2020/12/1.24/)
- [Hooks](http://localhost:4000/blog/2020/12/1.24/) for the React integration
- Support for picking files from [Box](/blog/2020/11/1.23/), [Unsplash](/blog/2020/10/1.22/), [Facebook](/blog/2020/04/1.10/), and [OneDrive](/blog/2020/04/1.10/) 
- [“Ghost” files](/blog/2021/05/1.29/), as part of a revamped Golden Retriever plugin, which makes recovering lost files even more intuitive
- Support for 25 more languages (bringing it to a total of [37](/docs/locales/#List-of-locale-packs))
- [Dark mode](/blog/2020/04/1.13/) 
- an [Image Editor](/blog/2020/07/1.18-image-editor/)
- And four new core team members: [@ajkachnic](https://github.com/ajkachnic), [@mifi](https://github.com/mifi), [@aduh95](https://github.com/aduh95), and [@murderlon](https://github.com/murderlon)

## Smaller bundles

> With 2.0, following in the footsteps of Microsoft, we are dropping support for IE11. As a result, we are able to remove all built-in polyfills, and the new bundle size is **25% smaller**! If you want your app to still support older browsers (such as IE11), you should add the following polyfills to your bundle:

Reviewer: ```suggestion
With 2.0, following in the footsteps of Microsoft, we are dropping support for IE11. As a result, we are able to remove all built-in polyfills, and the new bundle size is **25% smaller**! If you want your app to still support older browsers (such as IE11), you may need to add the following polyfills to your bundle:
```

This is now the new default. Thanks to [@martin-brennan](https://github.com/martin-brennan) for this contribution!

Do you care about reliable uploads? You could also consider [`@uppy/tus`][tus] with a self-hosted or Transloadit Tus server. Tus can resume uploads, supports smaller chunks, and offers increased upload speeds.

## And more

- The `.run` method on the `Uppy` instance has been removed. This method was already obsolete and only logged a warning. As of this major version, it no longer exists.

- [`@uppy/informer`][informer] now supports showing multiple notifications at the same time. The notifications themselves have also been improved.

- Improved screen reader accessibility for checkboxes and the 'remove file' button for [`@uppy/dashboard`][dashboard].

- Sort files and folders alphabetically in the Google Drive provider.

- Polished our code base with improved eslint rules, private field methods, and other modern JavaScript features that help us write more intention-revealing and safe code.

- To make Uppy more friendly towards new contributors, we have renamed our `master` branch to `main`.

## What future remains for 1.0?

> Uppy 1.0 will continue to receive bug fixes for three more months (until 1 December 2021), security fixes for one more year (until 1 September 2022), but no more new features after today. Exceptions are unlikely, but _can_ be made – to accommodate those with commercial support contracts, for example.

Reviewer: ```suggestion
Uppy 1.0 will continue to receive bug fixes for three more months (until <time datetime="2021-12-01">1 December 2021</time>), security fixes for one more year (until <time datetime="2022-09-01">1 September 2022</time>), but no more new features after today. Exceptions are unlikely, but _can_ be made – to accommodate those with commercial support contracts, for example.
```

- [And more](#And-more)
- [What future remains for 1.0](#What-future-remains-for-1-0)
- [That's it](#That’s-it)

## Highlights since 1.0

A lot of things have happened since 1.0. In case you have missed some of our 1.x releases, here is a small overview. We have added:

- Official integrations for [Vue](/blog/2020/11/1.23/) ([3](/blog/2021/04/1.27/)), [Angular](/blog/2021/07/1.30/), and [Svelte](/blog/2020/12/1.24/)
- [Hooks](http://localhost:4000/blog/2020/12/1.24/) for the React integration
- Support for picking files from [Box](/blog/2020/11/1.23/), [Unsplash](/blog/2020/10/1.22/), [Facebook](/blog/2020/04/1.10/), [OneDrive](/blog/2020/04/1.10/) 
- [“Ghost” files](/blog/2021/05/1.29/), as part of a revamped Golden Retriever Plugin, which makes recovering lost files more intuitive
- Support for 25 more languages (to a [total of 37](/docs/locales/#List-of-locale-packs))
- [Dark mode](/blog/2020/04/1.13/)
- [Image Editor](/blog/2020/07/1.18-image-editor/)
- Four new core team members: [@ajkachnic](https://github.com/ajkachnic), [@mifi](https://github.com/mifi), [@aduh95](https://github.com/aduh95), and [@murderlon](https://github.com/murderlon).

## Smaller bundles

> With 2.0, following Microsoft, we are dropping support for IE11. As a result, we are able to remove all built-in polyfills, and the new bundle size is 25% smaller! If you need your app to support older browsers (such as IE11), you should add the following polyfills to your bundle:

Reviewer: I wonder if we should link to the relevant guide instead, in case the list of polyfills is amended in a future semver-minor version. wdyt?

Author: I'm personally okay with leaving that to the changelog. Future versions could reference changes in polyfills. We don't officially support IE as of 2.0.0 so if people decide to use future releases it can be expected not to immediately work. Besides, maintaining a guide with polyfills would mean testing it with polyfills mentioned, which is essentially capacity spend on maintenance again.

Reviewer: Maybe we can rephrase this so it doesn't imply its an exhaustive list
```suggestion
With 2.0, following Microsoft, we are dropping support for IE11. As a result, we are able to remove all built-in polyfills, and the new bundle size is 25% smaller! If you need your app to support older browsers (such as IE11), you may need to add the following polyfills to your bundle:
```

## Faster

Uppy now loads faster due to the decreased bundle size, and compared to 1.0, adding many files (hundreds or thousands), used to take dozens of seconds, which has now been reduced to milliseconds. So for that use case, speed improvements of 1000X are not uncommon or exaggerated.

This was made possible by avoiding rerendering all the file components whenever something changes, using `memoize` and `virtual-list` (allows us to only render what is actually on-screen). In addition, multiple files are now added to state in one go via `uppy.addFiles(Array)` vs when `uppy.addFile(File)` + `uppy.setState` were being called in a loop.

Before optimizations:

![](/images/blog/2.0/uppy-performance-before.gif)

After:

![](/images/blog/2.0/uppy-performance-after.gif)

## Preact X and upgraded dependencies

We’ve upgraded almost all of Uppy’s dependencies. This includes the migration to Preact X. All plugins that depend on Preact have been upgraded from `8.2.9` to the latest version `10.5.13`. If you’d like your custom plugin to continue working with Uppy 2.0, it also needs to be using latest Preact now.

> Other notable upgrades include browserify to `v10`, typescript to `v4.3`, autoprefixer to `v10`, and lerna to `v4`.

Reviewer: Not sure users would or should care about lerna, AFAIK it's only relevant for Uppy development and doesn't affect users.
```suggestion
Other notable upgrades include browserify to `v10`, typescript to `v4.3`, and autoprefixer to `v10`.
```

Reviewer: @arturi could you apply this one also please?

This is now the new default. Thanks to [@martin-brennan](https://github.com/martin-brennan) for making this contribution.

Care about reliable uploads? You could also consider [`@uppy/tus`][tus] with a self-hosted or Transloadit Tus server. Tus can resume uploads, support smaller chunks, and higher upload speeds.

## And more

- The `.run` method on the `Uppy` instance has been removed. This method was already obsolete and only logged a warning. As of this major version it no longer exists.

- [`@uppy/informer`][informer] now supports showing multiple notifications at the same time. The notifications themselves have also been improved.

- Improved screenreader accessibility for checkboxes and the remove file button for [`@uppy/dashboard`][dashboard].

- Sort files and folders alphabetically in the Google Drive provider.

- Polished our code base with improved eslint rules, private field methods, and other modern JavaScript features that help us write more intention revealing and safe code.

- To make Uppy more friendly towards new contributors, we have renamed our `master` branch to `main`.

## What future remains for 1.0

> 1.0 will continue to receive bug fixes for 3 more months (until 1 December 2021), security fixes for 1 more year (until 1 September 2022), but, no new features as of today. Exceptions are unlikely but _can_ be made, for example to accomodate those with commercial support contracts.

Reviewer: nit: ISO-8901 / RFC 3339 date format
```suggestion
1.0 will continue to receive bug fixes for 3 more months (until 2021-12-01), security fixes for 1 more year (until 2022-09-01), and no new features as of today. Exceptions are unlikely but _can_ be made, for example to accomodate those with commercial support contracts.
```

Author: Readability wise I don't think this is an improvement

Reviewer: We should at least use [`<time>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time) then.
```suggestion
1.0 will continue to receive bug fixes for 3 more months (until <time datetime="2021-12-01">1 December 2021</time>), security fixes for 1 more year (until <time datetime="2022-09-01">1 September 2022</time>), and no new features as of today. Exceptions are unlikely but _can_ be made, for example to accomodate those with commercial support contracts.
```
